reactor-core/src/main/java/reactor/util/retry/Retry.java

```diff
@@ -0,0 +1,384 @@
+		 * <p>
+		 * For example:
+		 * <pre><code>
+		 * //given
+		 * Builder retryTwiceIllegalArgument = Retry.max(2)
+		 *     .throwablePredicate(e -> e instanceof IllegalArgumentException);
+		 *
+		 * Builder retryTwiceIllegalArgWithCause = retryTwiceIllegalArgument.throwablePredicate(old ->
+		 *     old.and(e -> e.getCause() != null));
+		 * </code></pre>
+		 *
+		 * @param predicateAdjuster a {@link Function} that returns a new {@link Predicate} given the
+		 * currently in place {@link Predicate} (usually deriving from the old predicate).
+		 * @return the builder for further configuration
+		 */
+		public Builder throwablePredicate(
+				Function<Predicate<? super Throwable>, Predicate<? super Throwable>> predicateAdjuster) {
+			Objects.requireNonNull(predicateAdjuster, "predicateAdjuster");
+			return new Builder(
+					this.maxAttempts,
+					predicateAdjuster.apply(throwablePredicate),
+					this.isTransientErrors,
+					this.minBackoff,
+					this.maxBackoff,
+					this.jitterFactor,
+					this.backoffScheduler);
+		}
+
+		/**
+		 * Set the transient error mode, indicating that the strategy being built should use
+		 * {@link RetrySignal#failureSubsequentIndex()} rather than {@link RetrySignal#failureTotalIndex()}.
+		 * Transient errors are errors that could occur in bursts but are then recovered from by
+		 * a retry (with one or more onNext signals) before another error occurs.
+		 * <p>
+		 * In simplified mode, this means that the {@link #maxAttempts(long)} is applied
+		 * to each burst individually. In exponential backoff, the backoff is also computed
+		 * based on the index within the burst, meaning the next error after a recovery will
+		 * be retried with a {@link #minBackoff(Duration)} delay.
+		 *
+		 * @param isTransientErrors {@code true} to activate transient mode
+		 * @return the builder for further configuration
+		 */
+		public Builder transientErrors(boolean isTransientErrors) {
+			return new Builder(
+					this.maxAttempts,
+					this.throwablePredicate,
+					isTransientErrors,
+					this.minBackoff,
+					this.maxBackoff,
+					this.jitterFactor,
+					this.backoffScheduler);
+		}
+
+		//all backoff specific methods should set the default scheduler if needed
+
+		/**
+		 * Set the minimum {@link Duration} for the first backoff. This method switches to an
+		 * exponential backoff strategy if not already done so. Defaults to {@link Duration#ZERO}
+		 * when the strategy was initially not a backoff one.
+		 *
+		 * @param minBackoff the minimum backoff {@link Duration}
+		 * @return the builder for further configuration
+		 */
+		public Builder minBackoff(Duration minBackoff) {
+			return new Builder(
+					this.maxAttempts,
+					this.throwablePredicate,
+					this.isTransientErrors,
+					Objects.requireNonNull(minBackoff, "minBackoff"),
+					this.maxBackoff,
+					this.jitterFactor,
+					this.backoffScheduler == null ? Schedulers.parallel() : this.backoffScheduler);
```

Comment:
default `backoffScheduler` should probably go to the build method, or at least be extracted into a constant, because it seems to be used in quite a few `*Backoff` methods and one may forget to change it everywhere